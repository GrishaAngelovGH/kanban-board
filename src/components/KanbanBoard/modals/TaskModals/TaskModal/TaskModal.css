.rsw-ce {
  max-height: 200px;
  overflow: auto;
}

.task-modal {
  height: 230px;
}