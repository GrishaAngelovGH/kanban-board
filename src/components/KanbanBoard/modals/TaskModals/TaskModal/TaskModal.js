import Modal from "components/Modal"

import Form from "react-bootstrap/Form"
import FormCheck from "react-bootstrap/FormCheck"

import Editor from "react-simple-wysiwyg"

import "./TaskModal.css"

const TaskModal = ({
  show, isEdit, modalTitle, title, description, priority, isTemplate, templates, showTemplates,
  onShowTemplatesChange, onTitleChange, onDescriptionChange,
  onPriorityChange, onTemplateChange, onClose, onConfirm
}) => (
  <Modal
    show={show}
    onClose={onClose}
    onConfirm={onConfirm}
    title={modalTitle}
    disabledConfirm={(!title.length && !isTemplate) || !description.length}
    body={
      <div className="row overflow-auto task-modal">
        <div className="col-md-12">
          {
            !isEdit && !showTemplates && (
              <FormCheck
                checked={isTemplate}
                label="Use as a template"
                className="mb-3"
                onChange={onTemplateChange}
              />
            )
          }

          {
            !isEdit && (<FormCheck
              checked={showTemplates}
              label="Create from a template"
              className="mb-3"
              onChange={onShowTemplatesChange}
            />
            )
          }

          {
            !isTemplate && !showTemplates && (
              <Form.Control
                value={title}
                placeholder="Title"
                className="mb-3"
                onChange={onTitleChange}
              />
            )
          }

          {
            !showTemplates && <Editor value={description} onChange={onDescriptionChange} />
          }

          {
            !isTemplate && !showTemplates && (
              <Form.Select value={priority} onChange={onPriorityChange} className="mt-3">
                <option value="" className="text-secondary fw-bold">No Priority</option>
                <option value="low" className="text-warning fw-bold">Low Priority</option>
                <option value="medium" className="text-success fw-bold">Medium Priority</option>
                <option value="high" className="text-danger fw-bold">High Priority</option>
              </Form.Select>
            )
          }

          {
            showTemplates && templates.map(v => (
              <div key={v.id} className="border border-3 rounded m-3 p-3 shadow">
                <p className="text-secondary">{v.description}</p>
              </div>
            ))
          }
        </div>
      </div>
    }
  />
)

export default TaskModal