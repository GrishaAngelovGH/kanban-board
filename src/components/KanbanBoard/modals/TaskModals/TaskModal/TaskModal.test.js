import { render } from "@testing-library/react"

import TaskModal from "./TaskModal"

test("should render TaskModal component", () => {
  const onUpdate = jest.fn()

  const view = render(
    <TaskModal
      show={true}
      modalTitle="Create New Task"
      title="Title Task"
      description="Title Description"
      priority="medium"
      onTitleChange={onUpdate}
      onDescriptionChange={onUpdate}
      onPriorityChange={onUpdate}
    />
  )

  expect(view).toMatchSnapshot()
})

test("should render TaskModal component as a template", () => {
  const onUpdate = jest.fn()
  const onTemplateChange = jest.fn()

  const view = render(
    <TaskModal
      show={true}
      modalTitle="Create New Task"
      title="Title Task"
      description="Title Description"
      priority="medium"
      isTemplate={true}
      onTitleChange={onUpdate}
      onDescriptionChange={onUpdate}
      onPriorityChange={onUpdate}
      onTemplateChange={onTemplateChange}
    />
  )

  expect(view).toMatchSnapshot()
})

test("should render TaskModal component in edit mode", () => {
  const onUpdate = jest.fn()
  const onTemplateChange = jest.fn()

  const view = render(
    <TaskModal
      show={true}
      isEdit={true}
      modalTitle="Create New Task"
      title="Title Task"
      description="Title Description"
      priority="medium"
      isTemplate={true}
      onTitleChange={onUpdate}
      onDescriptionChange={onUpdate}
      onPriorityChange={onUpdate}
      onTemplateChange={onTemplateChange}
    />
  )

  expect(view).toMatchSnapshot()
})