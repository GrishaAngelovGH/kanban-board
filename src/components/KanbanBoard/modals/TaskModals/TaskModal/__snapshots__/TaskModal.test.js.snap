// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TaskModal component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              Create New Task
            </div>
          </div>
          <div
            class="modal-body"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="mb-3 form-check"
                >
                  <input
                    class="form-check-input"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    title=""
                  >
                    Use as a template
                  </label>
                </div>
                <input
                  class="mb-3 form-control"
                  placeholder="Title"
                  value="Title Task"
                />
                <div
                  class="rsw-editor"
                >
                  <div
                    class="rsw-toolbar"
                  >
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Undo"
                      type="button"
                    >
                      ‚Ü∂
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Redo"
                      type="button"
                    >
                      ‚Ü∑
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Bold"
                      type="button"
                    >
                      ùêÅ
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Italic"
                      type="button"
                    >
                      ùë∞
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Underline"
                      type="button"
                    >
                      <span
                        style="text-decoration: underline;"
                      >
                        ùêî
                      </span>
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Strike through"
                      type="button"
                    >
                      <s>
                        ab
                      </s>
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Numbered list"
                      type="button"
                    >
                      <svg
                        height="20"
                        style="vertical-align: text-top;"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M6.99938 12.998v-2H20.9994v2H6.99938zm0 6.0001v-2H20.9994v2H6.99938zm0-12.00001v-2H20.9994v2H6.99938zm-4 1v-3h-1v-1h2v4h-1zm-1 9.00001v-1h3v4h-3v-1h2v-.5h-1v-1h1v-.5h-2zM4.25 10c.41421 0 .75.3358.75.75 0 .2024-.08017.3861-.2105.521L3.11983 13H5v1H2v-.9218L4 11H2v-1h2.25z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Bullet list"
                      type="button"
                    >
                      <svg
                        height="20"
                        style="vertical-align: text-top;"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 5h14v2H7V5zm0 8v-2h14v2H7zM4 4.50001c.83 0 1.5.66992 1.5 1.5 0 .83007-.67 1.5-1.5 1.5s-1.5-.66993-1.5-1.5c0-.83008.67-1.5 1.5-1.5zM4 10.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5-1.5-.67-1.5-1.5.67-1.5 1.5-1.5zM7 19v-2h14v2H7zm-3-2.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5-1.5-.67-1.5-1.5.67-1.5 1.5-1.5z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Link"
                      type="button"
                    >
                      üîó
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Clear formatting"
                      type="button"
                    >
                      TÃ≤‚Çì
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="HTML mode"
                      type="button"
                    >
                      &lt;/&gt;
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <select
                      class="rsw-dd"
                      title="Styles"
                    >
                      <option
                        hidden=""
                      >
                        Styles
                      </option>
                      <option
                        value="0"
                      >
                        Normal
                      </option>
                      <option
                        value="1"
                      >
                        ùóõùó≤ùóÆùó±ùó≤ùóø ùü≠
                      </option>
                      <option
                        value="2"
                      >
                        Header 2
                      </option>
                      <option
                        value="3"
                      >
                        ùô≤ùöòùöçùöé
                      </option>
                    </select>
                  </div>
                  <div
                    class="rsw-ce"
                    contenteditable="true"
                  >
                    Title Description
                  </div>
                </div>
                <select
                  class="mt-3 form-select"
                >
                  <option
                    class="text-secondary fw-bold"
                    value=""
                  >
                    No Priority
                  </option>
                  <option
                    class="text-warning fw-bold"
                    value="low"
                  >
                    Low Priority
                  </option>
                  <option
                    class="text-success fw-bold"
                    value="medium"
                  >
                    Medium Priority
                  </option>
                  <option
                    class="text-danger fw-bold"
                    value="high"
                  >
                    High Priority
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Close
            </button>
            <button
              class="btn btn-primary"
              type="button"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render TaskModal component as a template 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 2048px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              Create New Task
            </div>
          </div>
          <div
            class="modal-body"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="mb-3 form-check"
                >
                  <input
                    checked=""
                    class="form-check-input"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    title=""
                  >
                    Use as a template
                  </label>
                </div>
                <div
                  class="rsw-editor"
                >
                  <div
                    class="rsw-toolbar"
                  >
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Undo"
                      type="button"
                    >
                      ‚Ü∂
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Redo"
                      type="button"
                    >
                      ‚Ü∑
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Bold"
                      type="button"
                    >
                      ùêÅ
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Italic"
                      type="button"
                    >
                      ùë∞
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Underline"
                      type="button"
                    >
                      <span
                        style="text-decoration: underline;"
                      >
                        ùêî
                      </span>
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Strike through"
                      type="button"
                    >
                      <s>
                        ab
                      </s>
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Numbered list"
                      type="button"
                    >
                      <svg
                        height="20"
                        style="vertical-align: text-top;"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M6.99938 12.998v-2H20.9994v2H6.99938zm0 6.0001v-2H20.9994v2H6.99938zm0-12.00001v-2H20.9994v2H6.99938zm-4 1v-3h-1v-1h2v4h-1zm-1 9.00001v-1h3v4h-3v-1h2v-.5h-1v-1h1v-.5h-2zM4.25 10c.41421 0 .75.3358.75.75 0 .2024-.08017.3861-.2105.521L3.11983 13H5v1H2v-.9218L4 11H2v-1h2.25z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Bullet list"
                      type="button"
                    >
                      <svg
                        height="20"
                        style="vertical-align: text-top;"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 5h14v2H7V5zm0 8v-2h14v2H7zM4 4.50001c.83 0 1.5.66992 1.5 1.5 0 .83007-.67 1.5-1.5 1.5s-1.5-.66993-1.5-1.5c0-.83008.67-1.5 1.5-1.5zM4 10.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5-1.5-.67-1.5-1.5.67-1.5 1.5-1.5zM7 19v-2h14v2H7zm-3-2.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5-1.5-.67-1.5-1.5.67-1.5 1.5-1.5z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Link"
                      type="button"
                    >
                      üîó
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Clear formatting"
                      type="button"
                    >
                      TÃ≤‚Çì
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="HTML mode"
                      type="button"
                    >
                      &lt;/&gt;
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <select
                      class="rsw-dd"
                      title="Styles"
                    >
                      <option
                        hidden=""
                      >
                        Styles
                      </option>
                      <option
                        value="0"
                      >
                        Normal
                      </option>
                      <option
                        value="1"
                      >
                        ùóõùó≤ùóÆùó±ùó≤ùóø ùü≠
                      </option>
                      <option
                        value="2"
                      >
                        Header 2
                      </option>
                      <option
                        value="3"
                      >
                        ùô≤ùöòùöçùöé
                      </option>
                    </select>
                  </div>
                  <div
                    class="rsw-ce"
                    contenteditable="true"
                  >
                    Title Description
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Close
            </button>
            <button
              class="btn btn-primary"
              type="button"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render TaskModal component in edit mode 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 2048px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              Create New Task
            </div>
          </div>
          <div
            class="modal-body"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="rsw-editor"
                >
                  <div
                    class="rsw-toolbar"
                  >
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Undo"
                      type="button"
                    >
                      ‚Ü∂
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Redo"
                      type="button"
                    >
                      ‚Ü∑
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Bold"
                      type="button"
                    >
                      ùêÅ
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Italic"
                      type="button"
                    >
                      ùë∞
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Underline"
                      type="button"
                    >
                      <span
                        style="text-decoration: underline;"
                      >
                        ùêî
                      </span>
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Strike through"
                      type="button"
                    >
                      <s>
                        ab
                      </s>
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Numbered list"
                      type="button"
                    >
                      <svg
                        height="20"
                        style="vertical-align: text-top;"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M6.99938 12.998v-2H20.9994v2H6.99938zm0 6.0001v-2H20.9994v2H6.99938zm0-12.00001v-2H20.9994v2H6.99938zm-4 1v-3h-1v-1h2v4h-1zm-1 9.00001v-1h3v4h-3v-1h2v-.5h-1v-1h1v-.5h-2zM4.25 10c.41421 0 .75.3358.75.75 0 .2024-.08017.3861-.2105.521L3.11983 13H5v1H2v-.9218L4 11H2v-1h2.25z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Bullet list"
                      type="button"
                    >
                      <svg
                        height="20"
                        style="vertical-align: text-top;"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 5h14v2H7V5zm0 8v-2h14v2H7zM4 4.50001c.83 0 1.5.66992 1.5 1.5 0 .83007-.67 1.5-1.5 1.5s-1.5-.66993-1.5-1.5c0-.83008.67-1.5 1.5-1.5zM4 10.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5-1.5-.67-1.5-1.5.67-1.5 1.5-1.5zM7 19v-2h14v2H7zm-3-2.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5-1.5-.67-1.5-1.5.67-1.5 1.5-1.5z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Link"
                      type="button"
                    >
                      üîó
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="Clear formatting"
                      type="button"
                    >
                      TÃ≤‚Çì
                    </button>
                    <button
                      class="rsw-btn"
                      data-active="false"
                      title="HTML mode"
                      type="button"
                    >
                      &lt;/&gt;
                    </button>
                    <div
                      class="rsw-separator"
                    />
                    <select
                      class="rsw-dd"
                      title="Styles"
                    >
                      <option
                        hidden=""
                      >
                        Styles
                      </option>
                      <option
                        value="0"
                      >
                        Normal
                      </option>
                      <option
                        value="1"
                      >
                        ùóõùó≤ùóÆùó±ùó≤ùóø ùü≠
                      </option>
                      <option
                        value="2"
                      >
                        Header 2
                      </option>
                      <option
                        value="3"
                      >
                        ùô≤ùöòùöçùöé
                      </option>
                    </select>
                  </div>
                  <div
                    class="rsw-ce"
                    contenteditable="true"
                  >
                    Title Description
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Close
            </button>
            <button
              class="btn btn-primary"
              type="button"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
